core/psh/ps/ps.c

```diff
@@ -133,9 +155,9 @@ int psh_ps(int argc, char **argv)
 		printf("%6s ", buff);
 
 		if (!threads)
-			printf("%3u %-28s\n", info[i].tid, info[i].name);
+			printf("%3u %-*s\n", info[i].tid, (fullcmd ? sizeof(info[i].name) : 28), info[i].name);
 		else
-			printf("%-32s\n", info[i].name);
+			printf("%-*s\n", (fullcmd ? sizeof(info[i].name) : 32), info[i].name);
```

Comment:
`sizeof(info[i].name)` is always equal to 128 so when paired with very short commands it makes large gaps between ps entries when executed on short tty-s like qemu.
![Screenshot from 2021-06-01 13-29-37](https://user-images.githubusercontent.com/80880689/120317010-64868d80-c2de-11eb-84c3-bae92ae26318.png)
Maybe use `strlen()`? Does the job right, but it is has higher run cost.

Reply:
thanks for the remark, I've responded to the same issue reported by @Maxez above, I believe the new solution is good enough.